src/test/java/com/microsoft/azure/relay/SendReceiveTest.java

```diff
@@ -203,16 +205,17 @@ public void httpLargePOSTAndLargeResponseTest() throws IOException {
 	}
 	
 	private static CompletableFuture<Void> websocketClient(String msgExpected, String msgToSend) {
-		CompletableFuture<Boolean> receivedReply = new CompletableFuture<Boolean>();
+		AtomicBoolean receivedReply = new AtomicBoolean(false);
 		
 		return client.createConnectionAsync().thenAccept((channel) -> {
 			channel.writeAsync(StringUtil.toBuffer(msgToSend)).join();
 			channel.readAsync().thenAccept((bytesReceived) -> {
 				String msgReceived = new String(bytesReceived.array());
-				receivedReply.complete(true);
+				receivedReply.set(true);
 				assertEquals("Websocket sender did not receive the expected reply.", msgExpected, msgReceived);
+				channel.closeAsync().join();
```

Comment:
>join [](start = 25, length = 4)

instead of join here might it be better to change "thenAccept" to "thenCompose"? and `return channel.closeAsync();`?